scipy/sparse/data.py

```diff
@@ -64,8 +64,16 @@ def __itruediv__(self, other):  # self /= other
         else:
             return NotImplemented
 
-    def astype(self, t):
-        return self._with_data(self._deduped_data().astype(t))
+    def astype(self, dtype, casting='unsafe', copy=True):
+        dtype = np.dtype(dtype)
+        if self.dtype != dtype:
+            return self._with_data(
+                self._deduped_data().astype(dtype, casting=casting, copy=copy),
+                copy=False)
```

Comment:
You are right. So `copy=True` would be to correct choice. With `copy=copy` we could end up with matrices sharing the same indices arrays and different data arrays.